### PR TITLE
ENH: Composition and Balance types

### PR DESCRIPTION
**Pull request by mortonjt**
Adding in these types that are used in Gneiss, and other compositional stats tools.

### CODE REVIEW

**Comment by ebolyen** [comment]
I would probably put these variants in `q2_types.feature_table` since they are really only used in that context. (It will make importing them easier too, since you have to grab `FeatureTable` anyways)

**Comment by mortonjt** [comment]
Looks like we are getting type conflicts.  Has `q2-composition` been removed from the q2 workflow?  Or should we explicitly remove the `Composition` and `Balance` types from `q2-composition`?

File: q2_types/feature_table/_type.py [hunk 1]
```diff
@@ -22,11 +22,19 @@
...
+
+plugin.register_semantic_types(Composition, Balance)
 
 plugin.register_semantic_types(FeatureTable, Frequency, RelativeFrequency,
                                PresenceAbsence)
 
 plugin.register_semantic_type_to_format(
-    FeatureTable[Frequency | RelativeFrequency | PresenceAbsence],
+    FeatureTable[Frequency | RelativeFrequency |
+                 PresenceAbsence | Balance | Composition],
```
Comment (ebolyen) [inline q2_types/feature_table/_type.py]: :+1:

File: q2_types/feature_table/_type.py [hunk 2]
```diff
@@ -22,11 +22,19 @@
 PresenceAbsence = SemanticType('PresenceAbsence',
                                variant_of=FeatureTable.field['content'])
 
+Composition = SemanticType('Composition',
+                           variant_of=FeatureTable.field['content'])
+
+Balance = SemanticType('Balance',
+                       variant_of=FeatureTable.field['content'])
+
+plugin.register_semantic_types(Composition, Balance)
```
Comment (ebolyen) [inline q2_types/feature_table/_type.py]: I would just append these to the `register_semantic_types` on the next line down

**Comment by ebolyen** [comment]
@mortonjt, not yet but it will be. No need to touch q2-composition at this point.

**Pull request merged**